From 1748c8ffa487881c3e533b40b2ba3dfbd33e97cc Mon Sep 17 00:00:00 2001
From: Paul Lemire <paul.lemire@kdab.com>
Date: Mon, 18 Oct 2021 11:23:33 +0200
Subject: [PATCH 13/17] ObjectPicker: markDirty(AllDirty) when we change the
 enabled property

Otherwise we would simply change it internally but this wouldn't lead
to any actual changes in how we handle pick events.

Change-Id: I71aedc74db538978a939620d053af3167fd67dc4
Reviewed-by: Sean Harmer <sean.harmer@kdab.com>
(cherry picked from commit d32ff6b43b92474de495c88783a67b0b30211a7c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/render/picking/objectpicker.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/render/picking/objectpicker.cpp b/src/render/picking/objectpicker.cpp
index 182e1af55..cc2ebbf15 100644
--- a/src/render/picking/objectpicker.cpp
+++ b/src/render/picking/objectpicker.cpp
@@ -81,13 +81,16 @@ void ObjectPicker::syncFromFrontEnd(const Qt3DCore::QNode *frontEnd, bool firstT
     if (!node)
         return;
 
-    BackendNode::syncFromFrontEnd(frontEnd, firstTime);
-
     if (firstTime) {
         markDirty(AbstractRenderer::AllDirty);
         notifyJob();
     }
 
+    if (isEnabled() != node->isEnabled()) {
+        markDirty(AbstractRenderer::AllDirty);
+        // We let QBackendNode::syncFromFrontEnd change the enabled property
+    }
+
     if (node->isHoverEnabled() != m_hoverEnabled) {
         m_hoverEnabled = node->isHoverEnabled();
         markDirty(AbstractRenderer::AllDirty);
@@ -105,6 +108,8 @@ void ObjectPicker::syncFromFrontEnd(const Qt3DCore::QNode *frontEnd, bool firstT
         markDirty(AbstractRenderer::AllDirty);
         notifyJob();
     }
+
+    BackendNode::syncFromFrontEnd(frontEnd, firstTime);
 }
 
 void ObjectPicker::notifyJob()
-- 
2.36.1

